src/client/js/router.js

```diff
@@ -23,16 +23,19 @@ import highlightCategory from './highlight-category.js';
 /**
  * Router for smooth page tranistions.
  * Usage:
- * new Router();
+ * const router = new Router();
+ * router.enable();
  * It works as a progressive enhancement by hijacking the links
- * and popstate events. To disable page transitions simply don't
- * instantiate Router on page.
+ * and popstate events. To enable page transitions call enable()
+ * on instance of a Router.
  */
 class Router {
   constructor() {
     this._bindHandlers();
     this._hostname = location.host;
+  }
 
+  enable() {
     document.addEventListener('click', this._onLinkClick);
```

Comment:
Adding a `document` click listener is likely to cause unexpected side effects and potentially cause performance problems, isn't it? 

Every click will be channeled through `_onLinkClick()`.

Reply:
Yes, every click will be channeled through _onLinkClick().

What kind of problems do you expect?

Reply:
> What kind of problems do you expect?

Nothing specific, but I feel uneasy about code with global 'scope'. 

Adding a handler for every click on every element feels like something liable to become a future cause of unexpected side effects.

For example, when I was playing around with add to cart stuff, I got an error from the Router code that initially bamboozled me.

I guess we can keep it until there's evidence that it definitely causes a problem :).

Reply:
Another approach would be to change it to attach listener to each navigational link separately. It has disadvantages, as you're creating a dozen of listeners instead of 1, but is more intentional. Both have pros and cons. I put them in an issue so that we can discuss without blocking the PR: https://github.com/GoogleChromeLabs/sample-pie-shop/issues/85